include/deal.II/physics/elasticity/kinematics.h

```diff
@@ -0,0 +1,278 @@
+      * The result is expressed as
+      * @f[
+      *  \mathbf{w} := \frac{1}{2} \left[ \mathbf{l} - \mathbf{l}^{T} \right] \, .
+      * @f]
+      * where $\mathbf{l}$ is the spatial velocity gradient Tensor.
+      */
+     static Tensor<2, dim, Number>
+     w (const Tensor<2,spacedim> & F,
+        const Tensor<2,spacedim> & dF_dt);
+    };
+
+  }
+}
+
+
+
+// ------------------------- inline functions ------------------------
+
+
+
+template <int dim, typename Number>
+inline
+Tensor<2, dim, Number>
+Physics::Elasticity::Kinematics<dim,Number>::F (const Tensor<2, dim, Number> &Grad_u)
+{
+// See Wriggers (2008) p 23 equ. 3.14
+  return Physics::Elasticity::StandardTensors<dim>::I + Grad_u;
+}
+
+
+
+template <int dim, typename Number>
+inline
+SymmetricTensor<2, dim, Number>
+Physics::Elasticity::Kinematics<dim,Number>::C (const Tensor<2, dim, Number> &F)
+{
+// See Wriggers (2008) p 23 equ. 3.15 (text below)
+  return symmetrize(transpose(F)*F);
+}
+
+
+
+template <int dim, typename Number>
+inline
+SymmetricTensor<2, dim, Number>
+Physics::Elasticity::Kinematics<dim,Number>::b (const Tensor<2, dim, Number> &F)
+{
+// See Wriggers (2008) p 23 equ. 3.25 (text below)
+  return symmetrize(F*transpose(F));
+}
+
+
+
+template <int dim, typename Number>
+inline
+SymmetricTensor<2, dim, Number>
+Physics::Elasticity::Kinematics<dim,Number>::E (const Tensor<2, dim, Number> &F)
+{
+// See Wriggers (2008) p 23 equ. 3.15
+  return Number(0.5)*(C(F) - unit_symmetric_tensor<dim,Number>());
+}
+
+
+
+template <int dim, typename Number>
+inline
+SymmetricTensor<2, dim, Number>
+Physics::Elasticity::Kinematics<dim,Number>::e (const Tensor<2, dim, Number> &Grad_u)
+{
+// See Wriggers (2008) p 24 equ. 3.17
+// This is the equivalent to 0.5*symmetrize(Grad_u + transpose(Grad_u));
+  return symmetrize(Grad_u);
+}
+
+
+
+template <int dim, typename Number>
+inline
+SymmetricTensor<2, dim, Number>
+Physics::Elasticity::Kinematics<dim,Number>::a (const Tensor<2, dim, Number> &F)
+{
+  const Tensor<2, dim, Number> F_inv = invert(F);
+// See Wriggers (2008) p 30 equ. 3.35
+  return Number(0.5)*symmetrize(unit_symmetric_tensor<dim,Number>() - transpose(F_inv)*F_inv);
+}
+
+
+
+template <int dim, typename Number>
+inline
+Tensor<2, dim, Number>
+Physics::Elasticity::Kinematics<dim,Number>::l (const Tensor<2,spacedim> & F,
+   const Tensor<2,spacedim> & dF_dt)
+{
+  // See Holtzapfel (2007) eq 2.141
+  return dF_dt*invert(F);
+}
+
+
+
+template <int dim, typename Number>
+inline
+SymmetricTensor<2, dim, Number>
+Physics::Elasticity::Kinematics<dim,Number>::d (const Tensor<2,spacedim> & F,
+   const Tensor<2,spacedim> & dF_dt)
+{
+  // See Holtzapfel (2007) eq 2.148
+  return symmetrize(l(F,dF_dt));
+}
+
+
+
+template <int dim, typename Number>
+inline
+Tensor<2, dim, Number>
+Physics::Elasticity::Kinematics<dim,Number>::w (const Tensor<2,spacedim> & F,
+   const Tensor<2,spacedim> & dF_dt)
+{
+  // See Holtzapfel (2007) eq 2.149
+  return l(F,dF_dt) - dF,dF_dt) ;
```

Comment:
@maieneuro Whilst you're looking around, do think that there's any relevant functionality that I've missed?